lib/src/main/java/io/ably/lib/realtime/ChannelBase.java

```diff
@@ -586,7 +586,7 @@ public void run() {
     /* State changes provoked by ConnectionManager state changes. */
 
     public void setConnected(boolean reattachOnResumeFailure) {
-        if (reattachOnResumeFailure){
+        if (reattachOnResumeFailure && state.isReattachable()){
```

Comment:
`reattachOnResumeFailure` flag becomes `true` when `connectionId` changes, don't know why we unconditionally attach all channels in this case.